src/Microsoft.AspNetCore.Mvc.Razor/Internal/DependencyContextRazorViewEngineOptionsSetup.cs

```diff
@@ -80,7 +80,7 @@ private static void SetCompilationOptions(RazorViewEngineOptions options, Extens
         {
             var parseOptions = options.ParseOptions;
             parseOptions = parseOptions.WithPreprocessorSymbols(
-                parseOptions.PreprocessorSymbolNames.Concat(compilationOptions.Defines));
+                parseOptions.PreprocessorSymbolNames.Concat(compilationOptions.Defines ?? Enumerable.Empty<string>()));
```

Comment:
Breaking change in DependencyContext. It returns null now if an app has no defines (instead of an Empty sequence).